javascript/sdk/corvy.js

```diff
@@ -1,251 +1,354 @@
-        this._log(`Error in message loop: ${err.message}`);
-        await new Promise(r => setTimeout(r, 5000));
-      }
-    }
-  }
-
-  async _handleCommand(msg) {
-    const content = msg.content.toLowerCase();
-    for (const [prefix, handler] of this.commands.entries()) {
-      if (content.startsWith(prefix)) {
-        const argsText = msg.content.slice(prefix.length).trim();
-        await handler(msg, this, argsText);
-
-        return true;
-      }
-    }
-
-    return false;
-  }
-
-  // -------------------------------
-  // API Helpers
-  // -------------------------------
-
-  async _sendMsg(flockId, nestId, content) {
-    try {
-      this._log(`Sending message to flock:${flockId}, nest:${nestId} -> "${content}"`);
-      await this.client.post(`/flocks/${flockId}/nests/${nestId}/messages`, { content });
-    } catch (err) {
-      this.emit("error", new Error("Send message failed: " + err.message));
-      this._log(`Send message failed: ${err.message}`);
-    }
-  }
-
-  async getUserById(userId) {
-    try {
-      this._log(`Fetching user by ID: ${userId}`);
-      const response = await this.client.get(`/users/${userId}`);
-      return response.data;
-    } catch (err) {
-      this.emit("error", new Error("Get user by id failed: " + err.message));
-      this._log(`Get user by id failed: ${err.message}`);
-    }
-  }
-
-  async getUserByUsername(username) {
-    try {
-      this._log(`Fetching user by username: ${username}`);
-      const response = await this.client.get(`/users/by-username/${username}`);
-
-      return response.data;
-    } catch (err) {
-      this.emit("error", new Error("Get user by username failed: " + err.message));
-      this._log(`Get user by username failed: ${err.message}`);
-    }
-  }
-
-  async getFlockById(flockId) {
-    try {
-      this._log(`Fetching flock by id: ${flockId}`);
-      const response = await this.client.get(`/flocks/${flockId}`);
-
-      return response.data;
-    } catch (err) {
-      this.emit("error", new Error("Get flock by id failed: " + err.message));
-      this._log(`Get flock by id failed: ${err.message}`);
-    }
-  }
-
-  // -------------------------------
-  // Stuff?
-  // -------------------------------
-
-  get uptime() {
-    return this.startTime ? Date.now() - this.startTime : 0;
-  }
-
-  get formattedUptime() {
-    const ms = this.uptime;
-    const sec = Math.floor((ms / 1000) % 60);
-    const min = Math.floor((ms / (1000 * 60)) % 60);
-    const hr = Math.floor((ms / (1000 * 60 * 60)) % 24);
-    const days = Math.floor(ms / (1000 * 60 * 60 * 24));
-    const parts = [];
-    
-    if (days) parts.push(`${days}d`);
-    if (hr) parts.push(`${hr}h`);
-    if (min) parts.push(`${min}m`);
-    if (sec || parts.length === 0) parts.push(`${sec}s`);
-
-    return parts.join(" ");
-  }
-};
\ No newline at end of file
+        this._log(`📨 Message in nest ${nestId} from ${sender}: '${msgContent}'`);
+        this.emit('messageRaw', message);
```

Comment:
messageRaw should include bot messages if you're striving for parity with the Python SDK

Reply:
No clue what you mean by this.

Reply:
line 201 removes all bot messages, before messageRaw is emitted